akka-http-tests/src/test/java/akka/http/javadsl/server/directives/ParameterDirectivesTest.java

```diff
@@ -31,11 +31,6 @@ public void testStringParameterExtraction() {
       .assertStatusCode(200)
       .assertEntity("john");
 
-    route
-        .run(HttpRequest.create().withUri("/abc?stringParam=a%b"))
```

Comment:
Could we use the `Raw-Request-URI` header instead, here?

Reply:
invalid percent-encodings aren't allowed even in 'relaxed' mode, so I'm not sure what to test in that case?

Or do we think it's too restrictive to disallow invalid percent-encodings in 'relaxed' mode? (it's what we do for other parts of the URI as well...)

Reply:
Just to keep this test (and the one on the Scala side) which explicitly tests the response to invalid URIs.

Reply:
The problem here is we can no longer create a `Uri` object with that error: it will throw an InvalidUriException before we have a chance to send it to the `Route`... so the 'corresponding' test is now on the URI parsing side, https://github.com/akka/akka-http/pull/3003/files#diff-197042ca975154fb454f1aea34cd34d7R649

Reply:
With `Raw-Request-URI` you don't have to create a URI object on the client side. But maybe it still wouldn't work because `RouteTest` doesn't support `Raw-Request-URI`?

Reply:
Indeed it doesn't support it - I guess we could add that but I'm not sure how useful that really is.

Reply:
Ok, let's just remove it for now. Thanks for checking.